ilastik/applets/labeling/labelingImport.py

```diff
@@ -0,0 +1,534 @@
+        busy_dlg.setMaximum(100)
+        def close_busy_dlg(*args):
+            QApplication.postEvent(busy_dlg, QCloseEvent())
+    
+        # Load the data from file into our cache
+        # When it's done loading, close the progress dialog.
+        req = opCache.Output[:]
+        req.notify_finished( close_busy_dlg )
+        req.notify_failed( close_busy_dlg )
+        req.submit()
+        busy_dlg.exec_()
+
+        readData = req.result
+        
+        maxLabels = len(labelingSlots.labelNames.value)
+        unique_read_labels, readLabelCounts = numpy.unique(readData, return_counts=True)
+        labelInfo = (maxLabels, (unique_read_labels, readLabelCounts))
+        del readData
+    
+        # Ask the user how to interpret the data.
+        settingsDlg = LabelImportOptionsDlg( parent_widget,
+                                             fileNames, opMetadataInjector.Output,
+                                             labelingSlots.labelInput, labelInfo )
+
+        def handle_updated_axes():
+            # The user is specifying a new interpretation of the file's axes
+            updated_axisorder = str(settingsDlg.axesEdit.text())
+            metadata = opMetadataInjector.Metadata.value.copy()
+            metadata.axistags = vigra.defaultAxistags(updated_axisorder)
+            opMetadataInjector.Metadata.setValue( metadata )
+        settingsDlg.axesEdit.editingFinished.connect( handle_updated_axes )
+
+        dlg_result = settingsDlg.exec_()
+        if dlg_result != LabelImportOptionsDlg.Accepted:
+            return
+
+        # Get user's chosen label mapping from dlg
+        labelMapping = settingsDlg.labelMapping    
+
+        # Get user's chosen offsets.
+        # Offsets in dlg only include the file axes, not the 5D axes expected by the label input,
+        # so expand them to full 5D 
+        axes_5d = opReorderAxes.Output.meta.getAxisKeys()
+        tagged_offsets = collections.OrderedDict( zip( axes_5d, [0]*len(axes_5d) ) )
+        tagged_offsets.update( dict( zip( opMetadataInjector.Output.meta.getAxisKeys(), settingsDlg.imageOffsets ) ) )
+        imageOffsets = tagged_offsets.values()
+
+        # Expect import is subset
+        if (TinyVector(opReorderAxes.Output.meta.shape) > writeSeeds.meta.shape).any():
+            QMessageBox.critical(parent_widget, "Import shape too large",
+                                 "Import shape is not a subset of original input stack.")
+            return
+
+        # Expect x, y shape to match original shape
+        labels_tagged_shape = labelingSlots.labelInput.meta.getTaggedShape()
+        file_tagged_shape = opReorderAxes.Output.meta.getTaggedShape()
+        if (  labels_tagged_shape['x'] != file_tagged_shape['x']
+           or labels_tagged_shape['y'] != file_tagged_shape['y'] ):
+            QMessageBox.critical(parent_widget, "Shape does not match",
+                                 "X,Y shape must match original input stack.")
+            return
```

Comment:
@holstgr-kaust, Is there a particular reason that the XY shape of the imported label data must match the raw data?